File: examples/pppj2006.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# pppj 2006 paper examples

r = PolyRing.new( ZZ(), "x1,x2,x3", PolyRing.lex );
puts "Ring: " + str(r);
puts;


f = 3 * x1**2 * x3**4 + 7 * x2**5 - 61;

puts "f = " + str(f);
puts;

id = r.ideal( "", [f] );
puts "Ideal: " + str(id);
puts;

ri = r.ring;
puts "ri = " + str(ri);

pol = id.pset;
puts "pol = " + str(pol);
puts

pol = ri.parse( str(f) );
puts "pol = " + str(pol);
puts;

#startLog();

File: examples/pppj2006.py
#
# jython examples for jas.
# $Id$
#

import sys;

from jas import PolyRing, ZZ
from jas import startLog

# pppj 2006 paper examples

#r = Ring( "Z(x1,x2,x3) L" );
r = PolyRing( ZZ(), "x1,x2,x3", PolyRing.lex );
print "Ring: " + str(r);
print;

#unused:
ps = """
( 
 ( 3 x1^2 x3^4 + 7 x2^5 - 61 )
) 
""";

f = 3 * x1**2 * x3**4 + 7 * x2**5 - 61;
print "f = " + str(f);
print;

#id = r.ideal( ps );
id = r.ideal( "", [f] );
print "Ideal: " + str(id);
print;

#startLog();

pps = """
 3 x1^2 x3^4 + 7 x2^5 - 61
""";

ri = r.ring;
print "ri = " + str(ri);
print

pol = id.pset;
print "pol = " + str(pol);
print

pol = ri.parse( pps );
print "pol = " + str(pol);
print
